### ISSUE TITLE
Moving ~/.notion to ~/.config/notion.

### ISSUE THREAD

**Issue by Scupake** [raboof/notion#310]
If that's currently possible, please show me how to do it, since I can't find anything about it.
If it's not, I would like you to consider following the XDG spec.

**Comment by raboof** [comment]
Perhaps we should support both, to make it less painful for existing users?

Would you be interested in preparing a PR?

**Comment by raboof** [comment]
Currently, I wonder if adding it to the search path with the `-s` parameter would work?

**Comment by Scupake** [comment]
> Perhaps we should support both, to make it less painful for existing users?

Maybe make it check if ```~/.config/notion``` exists, if it does use that, if it doesn't it uses ```~/.notion```.

> Would you be interested in preparing a PR?

uhhh, what do you mean exactly? Adding the code? or just an empty WIP PR?

**Comment by raboof** [comment]
> > Perhaps we should support both, to make it less painful for existing users?
> 
> Maybe make it check if `~/.config/notion` exists, if it does use that, if it doesn't it uses `~/.notion`.

That makes sense to me. Looking for all resources in both locations might also be fine.

> > Would you be interested in preparing a PR?
> 
> uhhh, what do you mean exactly? Adding the code? or just an empty WIP PR?

Adding the code ;)

**Comment by Scupake** [comment]
Oh, well I'm not very good at C, but I'll try to.

**Comment by md1023** [comment]
Symbolic links work well on my Linux installations:
```
23:40:43 pwd
/home/mnikolaev
23:40:52 ls -al .notion
lrwxrwxrwx 1 mnikolaev mnikolaev 50 янв 15 20:04 .notion -> /home/mnikolaev/Documents/olivetti/configs/.notion/
```
Is that desirable or you want to get rid of `~/.notion` completely?

**Comment by Scupake** [comment]
I want to get rid of it completely, home dir clutter annoys me a lot

**Comment by wilhelmy** [comment]
* [ ] notionflux needs to be handled separately, `~/.notion/notionflux.hist` is currently hardcoded

Other than that, I'm fine with supporting both locations.

**Comment by wilhelmy** [comment]
I think to formally support `~/.config` properly, also the XDG_CONFIG_HOME variable needs to be handled. Also the history shouldn't go into ~/.config/notion but ~/.local/share/notion or something like that. Haven't read the full spec: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

**Comment by Scupake** [comment]
Yeah, it's good idea. But personally just supporting ```~/.config``` is enough for me.
I'll try to support the spec though, gonna take a while though since *I'm really bad at programming in general and have never used C before*.

**Comment by raboof** [comment]
> But just supporting `~/.config` is enough for me.

I think first 'simply' adding support for `~/.config` and leaving more sophisticated support for future PR's would be fine.

> gonna take a while though since _I'm really bad at programming in general and have never used C before._

If you run into any roadblocks feel free to ask! I think it's great that you're even considering this at all!

> Maybe rename the issue to "XDG specification support."?

All of XDG is very broad, I think it's fine to have separate issues about various parts.

**Comment by wilhelmy** [comment]
@Scupake if you need any help, I think I have enough time on my hands to give some IRC support/mentoring. Speaking from personal experience, messing with things you're kind of clueless about is a fun way to learn stuff, if that's what you're after.
